### PR TITLE
feat: Allow controlling sidecar environment variables

### PR DESCRIPTION
**Pull request by gnosek**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area tests

> /area proposals

**What this PR does / why we need it**:

Without this PR, the injected sidecars are pretty much forced to run with the default environment. This PR introduces two ways to control the sidecar environment:

1. Any changes to the environment in the top-level environment_variables section are propagated to sidecars
2. Each sidecar (mount section in the config) can have its own private environment variables that take preference over the top-level ones

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

This is https://github.com/falcosecurity/kilt/pull/81, resubmitted without changes

### CODE REVIEW

File: pkg/hocon/task.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package hocon
+
+import (
+	"github.com/falcosecurity/kilt/pkg/kilt"
```
Comment (pgcrooks-sysdig) [inline pkg/hocon/task.go:4]: Does this need updated to `agent-kilt`?

File: runtimes/cloudformation/go.mod [hunk 2]
```diff
@@ -42,3 +42,5 @@ require (
 	golang.org/x/sys v0.9.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+replace github.com/falcosecurity/kilt/pkg => ../../pkg
```
Comment (pgcrooks-sysdig) [inline runtimes/cloudformation/go.mod:46]:
```suggestion
replace github.com/agent-kilt/kilt/pkg => ../../pkg
```

Or something similar?

**Review by pgcrooks-sysdig — CHANGES_REQUESTED** [review]
Hopefully the renames will unblock the tests.

**Comment by gnosek** [comment]
Working on these right now. Yeah, something like this.

**Review by francesco-racciatti — APPROVED** [review]
lgtm

**Review by pgcrooks-sysdig — APPROVED** [review]
LGTM

**Pull request merged**